### PR TITLE
Convert punycode.js into an ES module

### PR DESCRIPTION
**Pull request by mjethani**
Similar rationale to #3844.

`punycode.js` is now an ES module.

### CODE REVIEW

**Comment by mjethani** [comment]
@gorhill I may not have got this one hundred percent right.

I ran the following manual test:

1. Built and reloaded the extension in Chrome
2. Loaded a web page (nytimes.com)
3. Opened the popup and then the settings page

I didn't see any errors on the chrome://extensions page.

File: src/whitelist.html [hunk 1]
```diff
@@ -43,7 +43,6 @@
 <script src="lib/codemirror/addon/scroll/annotatescrollbar.js"></script>
 <script src="lib/codemirror/addon/search/searchcursor.js"></script>
 <script src="lib/codemirror/addon/selection/active-line.js"></script>
-<script src="lib/punycode.js"></script>
```
Comment (mjethani) [inline src/whitelist.html:46]: I couldn't find anything here that uses this, but let me know if I've got it wrong.

Comment (gorhill) [reply src/whitelist.html:46]: Seems it's not being used, but this makes me realize it should be used, as the entries in _"Trusted sites"_ should be rendered as Unicode, which is not the case currently. Can be removed until I fix this in the future.

**Comment by mjethani** [comment]
@gorhill about `publicsuffixlist.js`, I looked at the [original package](https://github.com/gorhill/publicsuffixlist.js). You'll want to add ES modules support at some point and make it a hybrid package. I think [Option 3](https://2ality.com/2019/10/hybrid-npm-packages.html#option-3%3A-bare-import-esm%2C-deep-import-commonjs-with-backward-compatibility) would be best.

Aside from how to organize the code, the main choice is whether ES modules should be the default or CommonJS should be the default. If CommonJS is the default, it's backwards compatible, and you don't have to bump the major version number. If ES modules is the default: all the actual code is in the ES modules file, which means it can be copied here, and the CommonJS file is a wrapper that uses [`import()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import). Since `import()` returns a promise, the code is no longer backwards compatible (i.e. `require()` doesn't _immediately_ give you the object), and client code needs to account for this.

For the moment, it's probably OK to modify the file included here to make it an ES module. Later this same file can be copied over to the original repo to finish the work there.

Should I do this?

If we do this, we lose the last remaining property on the `globals` object as far as the Node.js package is concerned. After this it's only one or two more steps to lose `src/js/global.js` entirely.

**Comment by mjethani** [comment]
> For the moment, it's probably OK to modify the file included here to make it an ES module. Later this same file can be copied over to the original repo to finish the work there.

tl;dr let's make `publicsuffixlist.js` an ES module too.

**Comment by gorhill** [comment]
I looked into this recently to publish a package to NPM but I realized I had to merge changes between the official library and the one used by uBO, so I balked at this at the time as I had other stuff to do -- feel free to work on this if you want.
